heat/core/dndarray.py

```diff
@@ -730,6 +730,10 @@ def __getitem__(self, key: Union[int, Tuple[int, ...], List[int, ...]]) -> DNDar
             slices = [slice(None)] * (self.ndim - (len(kst) + len(kend)))
             key = kst + slices + kend
 
+        def __is_singular(key: any, axis: int, self_proxy: DNDarray = self_proxy) -> bool:
```

Comment:
sanitize axis

Reply:
maybe rename `self_proxy` to `torch_proxy`

Reply:
I think both depend on whether this code will ever be used non-internally. My reasoning is that, theoretically, `self_proxy` can be any torch.Tensor and the function will work, so why should it be named after `self`.